mujoco_warp/_src/passive.py

```diff
@@ -142,51 +128,189 @@ def _gravity_force(
     wp.atomic_add(qfrc_gravcomp_out[worldid], dofid, wp.dot(jac, force))
 
 
+def _box_fluid(opt: Option):
+  opt_wind = opt.wind
+  opt_viscosity = opt.viscosity
+  opt_density = opt.density
+
+  @nested_kernel
```

Comment:
This gets a bit subjective, but I think the bar might be higher for using a nested kernel.  For tiles we have no choice, but for here, we can pass the opts in as params.

If it led to a significant performance difference I would understand, but this kernel looks pretty cheap.

In particular, my understanding is that the performance hit comes from thread divergence.  e.g. some threads have viscosity and others don't.  I don't think that's the case here, no?

Reply:
changed `@nested_kernel` -> `@wp.kernel`

Reply:
after discussion with @erikfrey, the code is updated to have option settings be kernel inputs instead of kernel closure constants

Reply:
as a result, `wp.static` calls are removed since these option settings are not constants.

potential performance implications for kernel closure and `wp.static` in this context is something that could be investigated in the future